### PR TITLE
feat: AAVE, UNI-V2-DAI-WETH

### PR DESCRIPTION
**Pull request by brianmcmichael**
# Description

Spell Description:

COB Team Name or Author(s):

# Contribution Checklist

- [x] First commit title starts with COB Team Name and Collateral type (e.g. `___-TOKEN-X`)
- [x] Code approved
- [x] Tests approved
- [x] CI Tests pass

# Checklist

- [x] Every contract variable/method declared as public/external private/internal
- [x] Verify expiration (`4 days + 2 hours` monthly and `30 days` for the rest)
- [x] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [x] Validate all addresses used are in Kovan changelog
- [x] Deploy spell to kovan `SOLC_FLAGS="--optimize --optimize-runs=1" dapp --use solc:0.6.11 build --extract && dapp create DssSpell --gas=5000000 --gas-price="$(seth --to-wei 3 "gwei")"`
- [x] Ensure contract is verified on `kovan` etherscan
- [x] Change test to use kovan spell address and deploy timestamp
- [x] Keep `Kovan-DssSpell.sol` and `Kovan-DssSpell.t.sol` the same, but make a copy in `archive`
- [ ] `squash and merge` this PR

### CODE REVIEW

File: src/Kovan-DssSpell.sol [hunk 1]
```diff
@@ -88,222 +87,211 @@ contract SpellAction {
...
+        // !!!!!!!! Only if PIP_UNIV2DAIETH = Osm or PIP_UNIV2DAIETH = Median and hasn't been already whitelisted due a previous deployed ilk
+        LPOsmAbstract(PIP_UNIV2DAIETH).kiss(MCD_END);
+        // Set UNIV2DAIETH Osm in the OsmMom for new ilk
+        // !!!!!!!! Only if PIP_UNIV2DAIETH = Osm
+        OsmMomAbstract(OSM_MOM).setOsm(ILK_UNIV2DAIETH_A, PIP_UNIV2DAIETH);
+
+        // Set the UNIV2DAIETH-A debt ceiling
+        VatAbstract(MCD_VAT).file(ILK_UNIV2DAIETH_A, "line", 3 * MILLION * RAD);
+        // Set the UNIV2DAIETH-A dust
+        VatAbstract(MCD_VAT).file(ILK_UNIV2DAIETH_A, "dust", 100 * RAD);
```
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:239]: This value is supposed to be different in Kovan, right?

File: src/Kovan-DssSpell.t.sol [hunk 2]
```diff
@@ -752,144 +809,161 @@ contract DssSpellTest is DSTest, DSMath {
...
-        faucet.gulp(address(renbtc));
-        assertEq(renbtc.balanceOf(address(this)), faucetAmount);
+        lpPip.poke();
+        hevm.warp(now + 3601);
+        lpPip.poke();
+        spot.poke(ilk);
 
         // Check median matches pip.src()
-        // assertEq(pipRENBTC.src(), address(medRENBTC));
+        assertEq(lpPip.src(), address(lp)); // TODO: Check orbs
```
Comment (kmbarry1) [inline src/Kovan-DssSpell.t.sol:898]: Should we add the orb check?

File: src/Kovan-DssSpell.sol (hunk 1 above)
Comment (brianmcmichael) [reply src/Kovan-DssSpell.sol:239]: 100 in Kovan, 500 in Mainnet

File: src/Kovan-DssSpell.t.sol (hunk 2 above)
Comment (lucas-manuel) [reply src/Kovan-DssSpell.t.sol:898]: Yes we should:
```
orb0 = 0x91a8a02393cc21cee04b482b0e4aa26bfa1aa79c 
orb1 = 0x0e30f0fc91fdbc4594b1e2e5d64e6f1f94cab23d
```

Comment (brianmcmichael) [reply src/Kovan-DssSpell.t.sol]: added tests for these

File: src/Kovan-DssSpell.sol [hunk 3]
```diff
@@ -74,6 +72,7 @@ contract SpellAction {
     // A table of rates can be found at
     //    https://ipfs.io/ipfs/QmefQMseb3AiTapiAKKexdKHig8wroKuZbmLtPLv4u2YwW
     //
+    uint256 constant ONE_PERCENT_RATE   = 1000000000315522921573372069;
```
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:75]: Any reason we're not pulling these from rates.sol?

Comment (brianmcmichael) [reply src/Kovan-DssSpell.sol:75]: We can only do this in the tests. To do it in the spell it would deploy the whole Rates contract.

Comment (kmbarry1) [reply src/Kovan-DssSpell.sol:75]: Oh duh I missed that this was the actual spell.

File: src/Kovan-DssSpell.sol [hunk 4]
```diff
@@ -74,6 +72,7 @@ contract SpellAction {
...
     uint256 constant THREE_PERCENT_RATE = 1000000000937303470807876289;
```
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:76]: Looks like this one isn't used and can be removed.

File: src/Kovan-DssSpell.sol [hunk 5]
```diff
@@ -88,222 +87,209 @@ contract SpellAction {
         address ILK_REGISTRY = CHANGELOG.getAddress("ILK_REGISTRY");
         address FAUCET       = CHANGELOG.getAddress("FAUCET");
 
+        // Set the global debt ceiling
+        // + 20 M to fix an error introduced in previous spell not counting the previous 20M of ETH-B (before introducing IAM)
+        // + 10 M for AAVE-A
+        // +  3 M for UNIV2DAIETH-A
+        VatAbstract(MCD_VAT).file("Line", VatAbstract(MCD_VAT).Line() + 23 * MILLION * RAD + 10 * MILLION * RAD);
```
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:94]: Nit: it's a little inconsistent to express as 20 + 10 + 3 in the comment but 23 + 10 in the code.

Comment (brianmcmichael) [reply src/Kovan-DssSpell.sol]: fixed

File: src/Kovan-DssSpell.sol (hunk 4 above)
Comment (brianmcmichael) [reply src/Kovan-DssSpell.sol]: removed

**Review by kmbarry1 — APPROVED** [review]
LGTM

I haven't checked everything exhaustively, but IMO this is good enough for Kovan.

**Review by lucas-manuel — APPROVED** [review]
LGTM.

**Review by kmbarry1 — APPROVED** [review]

**Pull request merged**